pyfr/plugins/solutionwriter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolutionWriterPlugin(BasePlugin, H5Writer):
+    name = 'solutionwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, *args, **kwargs):
+        BasePlugin.__init__(self, intg, cfgsect)
```

Comment:
Just because there is not an explicit `__init__` method does not change the principle.  Translating how we handle `EulerElements` into your example:

```
class A(object):
    def __init__(self,a):
        print('A')

class B(object):
    def __init__(self, a):
        print('B')
        super().__init__(a)

class D(B, A):
    def __init__(self, a):
        print('D')
        super().__init__(a)

class E(A):
    def __init__(self, a):
        print('E')
        super().__init__(a)

print('Constructing D')
d = D(5)

print('Constructing E')
e = E(5)
```
Which works as expected.  The key is in the ordering of the classes when you perform the inheritance.  You can demonstrate this for yourself by switching `BaseFluidElements` with `BaseAdvectionElements` and seeing what happens.